# Thomas Hoffmann, EMBL Heidelberg, structures-it@embl.de, 2024/02
# morphosamplers requirement of pydantic <2 clashes whith napari's requirement.
# change pydantic requirement from version < 2 to >=2
# Accordingly, transform any imports to pydantic.v1
# Import psygnalV1 instead of psygnal. PsygnalV1 has a patch to enforce using 
# pydantic.v1 eventhough V2 is loaded.
diff -ru morphosamplers-0.0.10/PKG-INFO morphosamplers-0.0.10_bump_pydantic2/PKG-INFO
--- morphosamplers-0.0.10/PKG-INFO	2023-11-27 16:48:35.350763000 +0100
+++ morphosamplers-0.0.10_bump_pydantic2/PKG-INFO	2024-02-29 11:12:14.621599361 +0100
@@ -19,7 +19,7 @@
 Requires-Dist: einops
 Requires-Dist: numpy
-Requires-Dist: psygnal
+Requires-Dist: psygnalV1
-Requires-Dist: pydantic<2
+Requires-Dist: pydantic>=2
 Requires-Dist: scipy
 Requires-Dist: typing-extensions
 Provides-Extra: test
diff -ru morphosamplers-0.0.10/pyproject.toml morphosamplers-0.0.10_bump_pydantic2/pyproject.toml
--- morphosamplers-0.0.10/pyproject.toml	2023-11-27 16:48:18.000000000 +0100
+++ morphosamplers-0.0.10_bump_pydantic2/pyproject.toml	2024-02-29 11:12:06.188358266 +0100
@@ -27,7 +27,7 @@
     "einops",
     "numpy",
-    "psygnal",
+    "psygnalV1",
-    "pydantic<2",
+    "pydantic>=2",
     "scipy",
     "typing-extensions"
 ]
diff -ru morphosamplers-0.0.11/src/morphosamplers/core.py morphosamplers-0.0.11_pydanticV2/src/morphosamplers/core.py
--- morphosamplers-0.0.11/src/morphosamplers/core.py	2024-03-06 16:12:51.000000000 +0100
+++ morphosamplers-0.0.11_pydanticV2/src/morphosamplers/core.py	2024-05-07 10:53:55.475563518 +0200
@@ -1,7 +1,7 @@
 from abc import abstractmethod
 from typing import TypeVar, Protocol, Generic
 
-from pydantic.generics import GenericModel
+from pydantic.v1.generics import GenericModel
 
 M = TypeVar("M", bound="MorphoModel")
 S = TypeVar("S")
diff -ru morphosamplers-0.0.11/src/morphosamplers/models/path.py morphosamplers-0.0.11_pydanticV2/src/morphosamplers/models/path.py
--- morphosamplers-0.0.11/src/morphosamplers/models/path.py	2024-03-06 16:12:51.000000000 +0100
+++ morphosamplers-0.0.11_pydanticV2/src/morphosamplers/models/path.py	2024-05-07 10:53:55.475563518 +0200
@@ -1,5 +1,5 @@
 import numpy as np
-from pydantic import validator
+from pydantic.v1 import validator
 
 from morphosamplers.core import MorphoModel
 
diff -ru morphosamplers-0.0.11/src/morphosamplers/models/sphere.py morphosamplers-0.0.11_pydanticV2/src/morphosamplers/models/sphere.py
--- morphosamplers-0.0.11/src/morphosamplers/models/sphere.py	2024-03-06 16:12:51.000000000 +0100
+++ morphosamplers-0.0.11_pydanticV2/src/morphosamplers/models/sphere.py	2024-05-07 10:53:55.475563518 +0200
@@ -1,6 +1,6 @@
 from typing import Tuple
 
-from pydantic import validator
+from pydantic.v1 import validator
 
 from morphosamplers.core import MorphoModel
 
diff -ru morphosamplers-0.0.11/src/morphosamplers/models/surface.py morphosamplers-0.0.11_pydanticV2/src/morphosamplers/models/surface.py
--- morphosamplers-0.0.11/src/morphosamplers/models/surface.py	2024-03-06 16:12:51.000000000 +0100
+++ morphosamplers-0.0.11_pydanticV2/src/morphosamplers/models/surface.py	2024-05-07 10:53:55.475563518 +0200
@@ -1,7 +1,7 @@
 from typing import List
 
 import numpy as np
-from pydantic import validator
+from pydantic.v1 import validator
 
 from morphosamplers.core import MorphoModel
 
diff -ru morphosamplers-0.0.11/src/morphosamplers/sample_types.py morphosamplers-0.0.11_pydanticV2/src/morphosamplers/sample_types.py
--- morphosamplers-0.0.11/src/morphosamplers/sample_types.py	2024-03-06 16:12:51.000000000 +0100
+++ morphosamplers-0.0.11_pydanticV2/src/morphosamplers/sample_types.py	2024-05-07 10:53:55.475563518 +0200
@@ -1,5 +1,5 @@
 import numpy as np
-from pydantic import BaseModel, root_validator, validator
+from pydantic.v1 import BaseModel, root_validator, validator
 from typing import Union
 from typing_extensions import TypeAlias
 
diff -ru morphosamplers-0.0.11/src/morphosamplers/spline.py morphosamplers-0.0.11_pydanticV2/src/morphosamplers/spline.py
--- morphosamplers-0.0.11/src/morphosamplers/spline.py	2024-03-06 16:12:51.000000000 +0100
+++ morphosamplers-0.0.11_pydanticV2/src/morphosamplers/spline.py	2024-05-07 11:44:19.497952138 +0200
@@ -4,8 +4,8 @@
 from typing import Any, Dict, List, Optional, Tuple, Union
 
 import numpy as np
-from psygnal import EventedModel
-from pydantic import PrivateAttr, conint, root_validator, validator
+from psygnalV1 import EventedModel
+from pydantic.v1 import PrivateAttr, conint, root_validator, validator
 from scipy.interpolate import splev, splprep
 from scipy.spatial.transform import Rotation, Slerp
 
diff -ru morphosamplers-0.0.11/src/morphosamplers/surface_spline.py morphosamplers-0.0.11_pydanticV2/src/morphosamplers/surface_spline.py
--- morphosamplers-0.0.11/src/morphosamplers/surface_spline.py	2024-03-06 16:12:51.000000000 +0100
+++ morphosamplers-0.0.11_pydanticV2/src/morphosamplers/surface_spline.py	2024-05-07 11:44:34.138220384 +0200
@@ -5,8 +5,8 @@
 
 import einops
 import numpy as np
-from psygnal import EventedModel
-from pydantic import PrivateAttr, conint, root_validator, validator
+from psygnalV1 import EventedModel
+from pydantic.v1 import PrivateAttr, conint, root_validator, validator
 from scipy.interpolate import splev, splprep, interp1d
 from scipy.spatial.transform import Rotation
 
Only in morphosamplers-0.0.11_pydanticV2/src/morphosamplers: .surface_spline.py.swp
diff -ru morphosamplers-0.0.11/tests/test_spline_model.py morphosamplers-0.0.11_pydanticV2/tests/test_spline_model.py
--- morphosamplers-0.0.11/tests/test_spline_model.py	2024-03-06 16:12:51.000000000 +0100
+++ morphosamplers-0.0.11_pydanticV2/tests/test_spline_model.py	2024-05-07 10:53:55.474563492 +0200
@@ -2,7 +2,7 @@
 
 import numpy as np
 import pytest
-from pydantic import ValidationError
+from pydantic.v1 import ValidationError
 
 from morphosamplers.spline import NDimensionalSpline, Spline3D
 
Only in morphosamplers-0.0.11_pydanticV2: v2.patch
